src/gpu/nogpu.rs

```diff
@@ -1,61 +1,69 @@
 use super::error::{GpuError, GpuResult};
 use ec_gpu_gen::threadpool::Worker;
-use ff::PrimeField;
+use ff::{Field, PrimeField};
 use group::prime::PrimeCurveAffine;
 use std::marker::PhantomData;
 use std::sync::Arc;
 
-pub struct MultiexpKernel<E>(PhantomData<E>)
+use crate::gpu::GpuName;
+
+pub struct MultiexpKernel<G>(PhantomData<G>)
 where
-    E: Engine;
+    G: PrimeCurveAffine;
 
-impl<E> MultiexpKernel<E>
+impl<G> MultiexpKernel<G>
 where
-    E: Engine,
+    G: PrimeCurveAffine,
 {
     pub fn create(_: bool) -> GpuResult<Self> {
         Err(GpuError::GpuDisabled)
     }
 
-    pub fn multiexp<G>(
+    pub fn multiexp(
         &mut self,
         _: &Worker,
         _: Arc<Vec<G>>,
         _: Arc<Vec<<G::Scalar as PrimeField>::Repr>>,
         _: usize,
         _: usize,
-    ) -> GpuResult<<G as PrimeCurveAffine>::Curve>
+    ) -> GpuResult<G::Curve>
     where
         G: PrimeCurveAffine,
     {
         Err(GpuError::GpuDisabled)
     }
 }
 
-use pairing::Engine;
-
 macro_rules! locked_kernel {
```

Comment:
A documentation comment I think would be helpful to explain what this macro is doing.  I assume this is for compatibility with code in if GPU support is compiled out?